src/main/java/com/example/ajaxonboardingproject/controller/NatsMovieSessionController.kt

```diff
@@ -0,0 +1,88 @@
+package com.example.ajaxonboardingproject.controller
+
+import com.example.ajaxonboardingproject.model.MovieSession
+import com.example.ajaxonboardingproject.service.MovieSessionService
+import com.example.ajaxonboardingproject.service.proto.converter.MovieSessionConverter
+import io.nats.client.Connection
+import jakarta.annotation.PostConstruct
+import org.springframework.stereotype.Component
+
+@Component
+class NatsMovieSessionAddController(
+    private val movieSessionService: MovieSessionService,
+    private val movieSessionConverter: MovieSessionConverter,
+    private val natsConnection: Connection
+) {
+    @PostConstruct
+    fun listenToNatsSubject() {
+        val subject = "movieSession.add"
+        natsConnection.subscribe(subject)
+        val dispatcher = natsConnection.createDispatcher { message ->
+            val receivedMessage = MovieSessionOuterClass.MovieSession.parseFrom(message.data)
+            val response = add(receivedMessage)
+            natsConnection.publish(message.replyTo, response.toByteArray())
+        }
```

Comment:
I don't know if I am sure, but I have a suggestion to perhaps combine these two classes into one.
NatsMovieSessionAddController and NatsMovieSessionUpdateController

From what I see here, dispatcher is a handler in which you write the logic for what to do. So, maybe, correct me if I'm wrong, you can combine these classes by writing the choosing logic in the dispatcher.
```
@PostConstruct
    fun listenToNatsSubject() {
        val addSubject = "movieSession.add"
        val updateSubjectPattern = "movieSession.update.*"
        natsConnection.subscribe(addSubject)
        natsConnection.subscribe(updateSubjectPattern)
        val dispatcher = natsConnection.createDispatcher { message ->
            val receivedMessage = MovieSessionOuterClass.MovieSession.parseFrom(message.data)
            val requestId = message.subject.split(".").last()

            val response = when {
                message.subject == addSubject -> add(receivedMessage)
                message.subject.matches(updateSubjectPattern.toRegex()) -> update(requestId, receivedMessage)
                else -> null // Handle other some other subjects if needed
            }
            response?.let { natsConnection.publish(message.replyTo, it.toByteArray()) }
        }

        dispatcher.subscribe(addSubject)
        
        dispatcher.subscribe(updateSubjectPattern)
    }
```

Reply:
outdated